### PR TITLE
Adds page list as child of submenu

### LINKED ISSUE #46413: Navigation & Page List: Page List block can't live inside a submenu
## What problem does this address?
It is currently impossible to add a Page List block to a Navigation submenu:

https://user-images.githubusercontent.com/275961/206525733-28a7e340-325a-42f5-8248-be78f433a7b2.mp4

## What is your proposed solution?
It should be possible to add Page List to a navigation submenu block.

### PR DESCRIPTION
**Pull request by draganescu**
## What?

Fixes #46413 - allows the page list block to be used inside submenus of a navigation block.

## Why?

Because now that Page list is filterable on parent it can be used to create self updating sections of the navigation.

## How?

- enables page list as a child of navigation link and navigation submenu
- adds a transform from navigation link to page list
- adds an attribute to page list that specifies whether the block is nested in a submenu block or not
- fixes some problems of page list with styling
  - in the editor the color props from the navigation block were not passed down to all nested inner blocks but only to the 1st level
  - on the front end the color props from the navigation block were only applied to the 1st level of nested inner blocks
- copies over one of the changes in #46416 that aligns the off canvas editor with the list view in how it receives the list of blocks, changing from passing the inner blocks of the selected navigation block to a tree of client ids.

## Testing Instructions

1. make sure to have some pages that are nested in a hierarchy
2. in a new post
3. add a navigation block
4. in the navihation block add a link and select it
5. click the transform menu in the navigation link's toolbar
6. click page list
7. **there should be a page list with all the pages now in the submenu**
8. select the parent navigation block
9. in the inspector in the color panel set colors for background and text for overlays
10. Save
11. **In the editor the colors are respected**
12. **On the front end the page list renders well in the menu and the colors are respected**

### Testing Instructions for Keyboard

N/A

## Screenshots or screencast 

https://user-images.githubusercontent.com/107534/206705918-f70af1a0-f02c-422f-8c83-e15ce6521d45.mp4

### CODE REVIEW

File: packages/block-library/src/page-list/style.scss [hunk 1]
```diff
@@ -15,3 +15,13 @@
...
 	}
 }
+
+.wp-block-navigation .wp-block-navigation__submenu-container {
+	.wp-block-page-list {
+		display: flex;
+		flex-direction: column;
+		justify-content: flex-start;
+		align-items: flex-start;
+		background-color: inherit;
```
Comment (MaggieCabrera) [inline packages/block-library/src/page-list/style.scss]: Maybe we don't need this after the double ul is fixed.

**Comment by MaggieCabrera** [comment]
I keep seeing the issue where a theme that has colors applied to the navigation block won't see them for the submenu with a page list on it. The list items are all missing the classes that would apply them. I tested this with Disco.

<details>
Screen Capture on 2022-12-13 at 09-55-07
</details>

File: packages/block-library/src/page-list/style.scss (hunk 1 above)
Comment (MaggieCabrera) [reply packages/block-library/src/page-list/style.scss]: I don't see the class `.wp-block-page-list` on the frontend anymore:

Screenshot 2022-12-13 at 09 57 46

File: packages/block-library/src/navigation/edit/menu-inspector-controls.js [hunk 2]
```diff
@@ -60,7 +70,7 @@ const MenuInspectorControls = ( {
 					/>
 					{ isOffCanvasNavigationEditorEnabled ? (
 						<OffCanvasEditor
-							blocks={ innerBlocks }
+							blocks={ clientIdsTree }
```
Comment (getdave) [inline packages/block-library/src/navigation/edit/menu-inspector-controls.js]: I think this should be a separate PR. The potential for bugs is high.

**Comment by MaggieCabrera** [comment]
I had to restart my env cause I was still seeing the double ul. Now I don't but I'm getting this on the site editor and I'm not sure if it's unrelated.

```
Uncaught TypeError: event.target.classList is undefined
    onMouseMove use-in-between-inserter.js:51
```

It only happens when I hover and open the submenus on the editor.

File: packages/block-library/src/page-list/style.scss [hunk 3]
```diff
@@ -15,3 +15,13 @@
 		background-color: inherit;
 	}
 }
+
+.wp-block-navigation .wp-block-navigation__submenu-container {
```
Comment (MaggieCabrera) [inline packages/block-library/src/page-list/style.scss]: we can remove this

File: packages/block-library/src/page-list/editor.scss [hunk 4]
```diff
@@ -22,6 +22,16 @@
 	}
 }
 
+.wp-block-navigation .wp-block-navigation__submenu-container {
```
Comment (MaggieCabrera) [inline packages/block-library/src/page-list/editor.scss]: we don't need this either

File: packages/block-library/src/page-list-item/edit.js [hunk 5]
```diff
@@ -30,10 +30,95 @@ function useFrontPageId() {
...
+ * 1: Overlay custom colors (if submenu)
+ * 2: Overlay theme colors (if submenu)
+ * 3: Custom colors
+ * 4: Theme colors
+ * 5: Global styles
+ *
+ * @param {Object}  context
+ * @param {boolean} isSubMenu
+ */
+function getColors( context, isSubMenu ) {
```
Comment (getdave) [inline packages/block-library/src/page-list-item/edit.js:46]: We sholud share this code from a canonical location as it's liable to produce bugs. We can do that here on in a followup which we should create an Issue for now.

File: packages/block-library/src/page-list-item/edit.js [hunk 6]
```diff
@@ -30,10 +30,95 @@ function useFrontPageId() {
...
+				innerBlocksColors.customBackgroundColor
+			),
+			[ `has-${ innerBlocksColors.backgroundColor }-background-color` ]:
+				!! innerBlocksColors.backgroundColor,
+		} ),
+		style: {
+			color: innerBlocksColors.customTextColor,
+			backgroundColor: innerBlocksColors.customBackgroundColor,
+		},
+	} );
```
Comment (getdave) [inline packages/block-library/src/page-list-item/edit.js]: As above, if this code is duplicating the same in submenu block then we should share it from a single location. Same requirements as above 🙏

**Review by getdave — COMMENTED** [review]
Some 🙏. Otherwise looks good.

I'll defer to @MaggieCabrera and others on the styling side of things.

**Comment by scruffian** [comment]
I get this message in the editor: 
```Warning: validateDOMNesting(...): <li> cannot appear as a descendant of <li>.```

**Comment by scruffian** [comment]
In the page list, if there are submenus, they don't open unless I select the page list block itself, which feels a bit strange.

**Comment by scruffian** [comment]
Another thing I noticed - the overlay colors don't always seem to be inherited:

<details>
Screenshot 2022-12-15 at 15 04 20
</details>

**Review by scruffian — APPROVED** [review]
This is looking good. Let's make sure we follow up with the refactoring PRs.

File: packages/block-library/src/page-list-item/edit.js (hunk 6 above)
Comment (draganescu) [reply packages/block-library/src/page-list-item/edit.js]: I think we should do a refactoring PR too for this thing. I think @scruffian made a separate issue for this.
